Tests/MeiliSearchUnitTests/TasksTests.swift

```diff
@@ -0,0 +1,52 @@
+@testable import MeiliSearch
+import XCTest
+
+// swiftlint:disable force_try
+class TasksTests: XCTestCase {
+  private var client: MeiliSearch!
+  private var index: Indexes!
+  private let uid: String = "movies_test"
+  private let session = MockURLSession()
+
+  override func setUp() {
+    super.setUp()
+    client = try! MeiliSearch(host: "http://localhost:7700", apiKey: "masterKey", session: session)
+    index = client.index(self.uid)
+  }
+
+  func testGetTasksWithParametersFromClient() {
+    let jsonString = """
+      {
+        "results": [],
+        "limit": 20,
+        "from": 5,
+        "next": 98
+      }
+      """
+
+    // Prepare the mock server
+    session.pushData(jsonString)
+
+    // Start the test with the mocked server
+    let expectation = XCTestExpectation(description: "Get keys with parameters")
+
+    self.client.getTasks(params: TasksQuery(limit: 20, from: 5, next: 98, types: ["indexCreation"])) { result in
```

Comment:
Maybe it is worth trying out with `indexUid: ["something"]` as well to ensure that the method called from the client is entering the correct`getAll`

Reply:
The same here: https://github.com/meilisearch/meilisearch-swift/pull/315#discussion_r936756875